src/transformers/models/univnet/feature_extraction_univnet.py

```diff
@@ -0,0 +1,461 @@
+            extractors such as [`SpeechT5FeatureExtractor`] which take the `win_length` in ms.
+        win_function (`str`, *optional*, defaults to `"hann_window"`):
+            Name for the window function used for windowing, must be accessible via `torch.{win_function}`
+        filter_length (`int`, *optional*, defaults to 1024):
+            The number of FFT components to use. If `None`, this is determined using
+            `transformers.audio_utils.optimal_fft_length`.
+        max_length_s (`int`, *optional*, defaults to 10):
+            The maximum input lenght of the model in seconds. This is used to pad the audio.
+        fmin (`float`, *optional*, defaults to 0.0):
+            Minimum mel frequency in Hz.
+        fmax (`float`, *optional*, defaults to `None`):
+            Maximum mel frequency in Hz.
+        mel_floor (`float`, *optional*, defaults to 1e-9):
+            Minimum value of mel frequency banks.
+        center (`bool`, *optional*, default to `False`):
+            Whether to pad the waveform so that frame `t` is centered around time `t * hop_length`. If `False`, frame
+            `t` will start at time `t * hop_length`.
+        compression_factor (`float`, *optional*, defaults to 1.0):
+            The multiplicative compression factor for dynamic range compression during spectral normalization.
+        compression_clip_val (`float`, *optional*, defaults to 1e-5):
+            The clip value applied to the waveform before applying dynamic range compression during spectral
+            normalization.
+        normalize_min (`float`, *optional*, defaults to -11.512925148010254):
+            The min value used for Tacotron 2-style linear normalization. The default is the original value from the
+            Tacotron 2 implementation.
+        normalize_max (`float`, *optional*, defaults to 2.3143386840820312):
+            The max value used for Tacotron 2-style linear normalization. The default is the original value from the
+            Tacotron 2 implementation.
+        model_in_channels (`int`, *optional*, defaults to 64):
+            The number of input channels to the [`UnivNetGan`] model. This should match
+            `UnivNetGan.config.model_in_channels`.
+        pad_end_length (`int`, *optional*, defaults to 10):
+            If padding the end of the spectrograms, the number of frames to append to the end of each spectrogram.
+    """
+
+    model_input_names = ["input_features", "attention_mask"]
+
+    def __init__(
+        self,
+        feature_size: int = 1,
+        sampling_rate: int = 24000,
+        padding_value: float = 0.0,
+        do_normalize: bool = False,
+        num_mel_bins: int = 100,
+        hop_length: int = 256,
+        win_length: int = 1024,
+        win_function: str = "hann_window",
+        filter_length: Optional[int] = 1024,
+        max_length_s: int = 10,
```

Comment:
For (1), if my understanding is correct, the current UnivNet implementation follows the [Whisper convention](https://github.com/huggingface/transformers/blob/2aef9a96011133f6b399b598fd69cfeca936eb37/src/transformers/models/whisper/feature_extraction_whisper.py#L47-L48) that `hop_length` (and related quantities like `win_length`) is expressed in terms of number of samples, rather than milliseconds like in e.g. [SpeechT5](https://github.com/huggingface/transformers/blob/2aef9a96011133f6b399b598fd69cfeca936eb37/src/transformers/models/speecht5/feature_extraction_speecht5.py#L56-L57). (See https://github.com/huggingface/transformers/pull/24799#discussion_r1315632056 for more discussion; for example, `hop_length` would not be an integer under default UnivNet settings when expressed in ms, which is why I've chosen to express it in terms of samples.)

With respect to `self.n_samples`, I believe the closest analogue in UnivNet is `self.num_max_samples`:

 https://github.com/huggingface/transformers/blob/7fc3f69d185c80d5eed5fe49c556819ffc34cc08/src/transformers/models/univnet/feature_extraction_univnet.py#L143

where `max_length_s` is equivalent to `chunk_length` in Whisper. This follows the convention used in CLAP:

https://github.com/huggingface/transformers/blob/2f3ea08a077ba3133fa8a604b22436cad250b055/src/transformers/models/clap/feature_extraction_clap.py#L53-L54

I guess the naming conventions can be made to match that of Whisper; I prefer the `self.num_max_samples` and `max_length_s` used in CLAP rather than `self.n_samples` and `chunk_length` used in Whisper because the former are more descriptive than the latter.

Reply:
Thanks for the great explanation! That's our bad for the inconsistencies in the repo 😅 Indeed in this case you're free to choose the one that is most relevant to UnivNet - so here happy to select the arg names from CLAP/SpeechT5.